volumina/pixelpipeline/imagesources.py

```diff
@@ -420,8 +423,95 @@ def toImage( self ):
                 #FIXME: maybe this should be done in a better way using an operator before the colortable request which properly handles 
                 #this problem 
                 warnings.warn("Data for colortable layers cannot be float, casting",RuntimeWarning)
-                a=a.astype(np.int32)
-            vigra.colors.applyColortable(a, self._colorTable, byte_view(img))
+                a = np.asanyarray(a, dtype=np.uint32)
+
+            # If we have a masked array with a non-trivial mask, ensure that mask is made transparent.
+            _colorTable = self._colorTable
+            if np.ma.is_masked(a):
```

Comment:
So, I think something got lost in transmission. Maybe this won't be totally clear until I submit changes, but here is my attempt to clear it up in writing.

First, as a utility function, it will be an optional piece of code that people can chose to use if they wish. It will not be something happening behind the scenes without anyone’s control.

Second, the only place I am planning to use this utility function is for ensuring the masked part of the array appears transparent, in other words, in place of lines 431-512. In this case, the case of masked arrays, a transparent layer is absolutely necessary. Further, this would leave other unmasked layers (i.e. ColortableImageLayer instances) untouched.

If something is still not clear, I'll blame it on that long portion of code and we can try again once we have a better piece of code to look at.